packages/core/strapi/lib/core-api/controller/index.d.ts

```diff
@@ -1,6 +1,6 @@
 import { Context } from 'koa';
 
-type Response = object;
+type ControllerResponse <T=unknown> = T | Promise<T>;
```

Comment:
A controller can also return undefined (no return) and set the response on the ctx directly

Reply:
Very valid point, thanks!